java/arcs/sdk/Entity.kt

```diff
@@ -16,15 +16,27 @@ typealias URL = String
 /** Wraps a ByteArray whose final byte is set to 0. */
 inline class NullTermByteArray(val bytes: ByteArray)
 
-abstract class Entity<T> {
+abstract class Entity {
     var internalId = ""
     abstract fun schemaHash(): String
-    abstract fun decodeEntity(encoded: ByteArray): T?
     abstract fun encodeEntity(): NullTermByteArray
     abstract fun isSet(): Boolean
     abstract fun getFieldsNotSet(): List<String>
 }
 
+/**
+ * Spec for an [Entity] type. Can create and deserialize new entities.
+ *
+ * Implementation classes are autogenerated for each entity type.
+ */
+interface EntitySpec<T : Entity> {
```

Comment:
Could be left for a later PR, but I feel like the time has come to split this file into separate files.  Things like StringDecoder/Encoder and NullTermByteArray shouldn't really be in here, IMO.

Reply:
I'm working on it in a follow up PR. It's part of my effort to pull all of the wasm-only code into the wasm package.